Cabal/Distribution/Types/ModuleRenaming.hs

```diff
@@ -87,18 +104,21 @@ instance Parsec ModuleRenaming where
     -- NB: try not necessary as the first token is obvious
     parsec = P.choice [ parseRename, parseHiding, return DefaultRenaming ]
       where
+        lparen = P.char '(' >> liberalSpaces
+        rparen = P.char ')' >> liberalSpaces
```

Comment:
@phadej The liberalSpaces parser just checks for appropriate cabal version and then skips spaces. For lower versions of the cabal spec it says gives an error if there is space. So far so good but I noticed that the spaces stuff was not needed in the version 2.2 case for rparen as there is a P.spaces after its call. Further down the code in the hiding clause there is no P.spaces used which means that the rparen there does not skip spaces (in the version 2.2 behaviour) 

If we want to maintain complete bug compatibility, then this will lead to a rather convoluted rparen behaviour. I feel this will eventually lead to more confusing whitespace behaviour in the end and the code will be pepperd with
various flavors of paren-parsers (some that skips spaces and some that do not).

Reply:
The question therefore is shall I replace the rparen definition with

```
rparen = P.char ')' >> P.spaces
```
This is not really compatible with V2.2 I think due to the use of rparen in the hiding clause.